sootup.core/src/main/java/sootup/core/graph/MutableStmtGraph.java

```diff
@@ -67,17 +67,27 @@ public void addBlock(@Nonnull List<Stmt> stmts) {
    */
   public abstract void replaceNode(@Nonnull Stmt oldStmt, @Nonnull Stmt newStmt);
 
-  public abstract void insertBefore(
+  public abstract BasicBlock<?> insertBefore(
       @Nonnull Stmt beforeStmt,
       @Nonnull List<FallsThroughStmt> stmts,
       @Nonnull Map<ClassType, Stmt> exceptionMap);
 
+  public abstract BasicBlock<?> insertAfter(
+      @Nonnull Stmt afterStmt,
+      @Nonnull List<FallsThroughStmt> stmts,
+      @Nonnull Map<ClassType, Stmt> exceptionMap);
+
   /**
    * inserts the "newStmt" before the position of "beforeStmt" i.e.
    * newStmt.successors().contains(beforeStmt) will be true
    */
-  public void insertBefore(@Nonnull Stmt beforeStmt, @Nonnull FallsThroughStmt newStmt) {
-    insertBefore(beforeStmt, Collections.singletonList(newStmt), Collections.emptyMap());
+  public BasicBlock<?> insertBefore(@Nonnull Stmt beforeStmt, @Nonnull FallsThroughStmt newStmt) {
+    return insertBefore(beforeStmt, Collections.singletonList(newStmt), Collections.emptyMap());
+  }
+
+  /** inserts the "newStmt" after the position of "afterStmt" */
+  public BasicBlock<?> insertAfter(@Nonnull Stmt afterStmt, @Nonnull FallsThroughStmt newStmt) {
+    return insertAfter(afterStmt, Collections.singletonList(newStmt), Collections.emptyMap());
```

Comment:
which behaviour do we want:
a) no exception for the new stmt (current)
b) fit in seamlessly  into existing block/surrounding structure (of the anchoring Stmt i.e. afterStmt)

Reply:
Behavior a. That means the new stmt is not in a trap.  If the anchoring stmt is also not in a  trap, then the behavior is just like b. If the anchoring stmt is in a trap, the new stmt and anchoring stmt will be in two different blocks.